### PR TITLE
Small refactor copy

### PR DESCRIPTION
**Pull request by alex-hse-repository**
The same as "small_refactor" branch with solved conflicts

### CODE REVIEW

File: etna/metrics/base.py [hunk 1]
```diff
@@ -30,7 +36,6 @@ def __init__(
     ):
         """
         Init Metric.
-
```
Comment (Mr-Geekman) [inline etna/metrics/base.py:33]: Leave this gap.

File: etna/metrics/base.py [hunk 2]
```diff
@@ -10,17 +11,22 @@
...
+    def _missing_(cls, value):
+        raise NotImplementedError(
+            f"{value} is not a valid {cls.__name__}. Only {', '.join([repr(m.value) for m in cls])} aggregation allowed"
+        )
+
 
 class Metric(BaseMixin):
     """
     Base class for all the multi-segment metrics.
-
```
Comment (Mr-Geekman) [inline etna/metrics/base.py:23]: Leave this gap.

File: etna/metrics/base.py [hunk 3]
```diff
@@ -44,29 +49,22 @@ def __init__(
...
-            raise NotImplementedError(
-                f"Only '{MetricAggregationMode.macro}' and '{MetricAggregationMode.per_segment}' aggregation allowed"
-            )
         self.mode = mode
 
     @staticmethod
     def _validate_segment_columns(y_true: TSDataset, y_pred: TSDataset):
         """
         Check if all the segments from y_true are in y_pred and vice versa.
-
```
Comment (Mr-Geekman) [inline etna/metrics/base.py:62]: Gap.

File: etna/metrics/base.py [hunk 4]
```diff
@@ -44,29 +49,22 @@ def __init__(
...
         Parameters
         ----------
         y_true:
             y_true dataset
         y_pred:
             y_pred dataset
-
```
Comment (Mr-Geekman) [inline etna/metrics/base.py:69]: Gap.

File: etna/metrics/base.py [hunk 5]
```diff
@@ -99,14 +97,12 @@ def _validate_segment_columns(y_true: TSDataset, y_pred: TSDataset):
     def _validate_timestamp_columns(timestamp_true: pd.Series, timestamp_pred: pd.Series):
         """
         Check that y_true and y_pred have the same timestamp.
-
```
Comment (Mr-Geekman) [inline etna/metrics/base.py:102]: Gap.

File: etna/metrics/base.py [hunk 6]
```diff
@@ -99,14 +97,12 @@ def _validate_segment_columns(y_true: TSDataset, y_pred: TSDataset):
...
         Parameters
         ----------
         timestamp_true:
             y_true's timestamp column
         timestamp_pred:
             y_pred's timestamp column
-
```
Comment (Mr-Geekman) [inline etna/metrics/base.py:109]: Gap.

File: etna/metrics/base.py [hunk 7]
```diff
@@ -119,11 +115,9 @@ def _validate_timestamp_columns(timestamp_true: pd.Series, timestamp_pred: pd.Se
     def _macro_average(metrics_per_segments: Dict[str, float]) -> float:
         """
         Compute macro averaging of metrics over segment.
-
```
Comment (Mr-Geekman) [inline etna/metrics/base.py:122]: Gap.

File: etna/metrics/base.py [hunk 8]
```diff
@@ -119,11 +115,9 @@ def _validate_timestamp_columns(timestamp_true: pd.Series, timestamp_pred: pd.Se
...
         Parameters
         ----------
         metrics_per_segments: dict of {segment: metric_value} for segments to aggregate
-
```
Comment (Mr-Geekman) [inline etna/metrics/base.py:126]: Gap.

File: etna/metrics/base.py [hunk 9]
```diff
@@ -133,18 +127,15 @@ def _macro_average(metrics_per_segments: Dict[str, float]) -> float:
     def __call__(self, y_true: TSDataset, y_pred: TSDataset) -> Union[float, Dict[str, float]]:
         """
         Compute metric's value with y_true and y_pred.
-
```
Comment (Mr-Geekman) [inline etna/metrics/base.py:136]: Gap.

File: etna/metrics/base.py [hunk 10]
```diff
@@ -133,18 +127,15 @@ def _macro_average(metrics_per_segments: Dict[str, float]) -> float:
...
         Notes
         -----
         Note that if y_true and y_pred are not sorted Metric will sort it anyway
-
```
Comment (Mr-Geekman) [inline etna/metrics/base.py:140]: Gap.

File: etna/metrics/base.py [hunk 11]
```diff
@@ -133,18 +127,15 @@ def _macro_average(metrics_per_segments: Dict[str, float]) -> float:
...
         Parameters
         ----------
         y_true:
             dataset with true time series values
         y_pred:
             dataset with predicted time series values
-
```
Comment (Mr-Geekman) [inline etna/metrics/base.py:147]: Gap.

**Review by Mr-Geekman — APPROVED** [review]

**Review by iKintosh — APPROVED** [review]
👍

**Pull request merged**